### PR TITLE
Add RelWithDebInfo & MinSizeRel build types

### PR DESCRIPTION
**Pull request by BenDol**
## 🍰 Pullrequest
Add RelWithDebInfo & MinSizeRel build types to CMAKE build.

### How2Test
Run CMake

### CODE REVIEW

**Comment by cyberium** [comment]
Merged Thank you.

File: CMakeLists.txt [hunk 1]
```diff
@@ -318,6 +318,8 @@ configure_file(
 )
 
 set(DEFINITIONS_RELEASE NDEBUG)
+set(DEFINITIONS_RELWITHDEBINFO NDEBUG _RELWITHDEBINFO MANGOS_RELWITHDEBINFO)
+set(DEFINITIONS_MINSIZEREL NDEBUG _MINSIZEREL MANGOS_MINSIZEREL)
```
Comment (evil-at-wow) [inline CMakeLists.txt]: The NDEBUG here is useful, but is there any point in adding MANGOS_RELWITHDEBINFO, MANGOS_MINSIZEREL,  _RELWITHDEBINFO and _MINSIZEREL? These are not used in code - and hopefully will never be - so why would we want to pass those to the compiler?

Comment (cyberium) [reply CMakeLists.txt]: For me those might be used in perfect world to add/remove some code needed in those scenario. We will probably never use them but well we can not say this is not a good code.

Comment (evil-at-wow) [reply CMakeLists.txt]: Actually we can say it's not good code (well, partly). Names starting with an underscore (_) followed by an uppercase letter are **reserved** for the implementation (read: your compiler). So using _RELWITHDEBINFO and _MINSIZEREL in code is a very big no-go for us. 

I also don't see what code you might ever want to add or remove in these cases (what would you possibly not want in debug or release, yet want in release with debug symbols?), but OK, if you really want it, maybe we can just keep the MANGOS_ ones and only those then.

**Pull request closed without merging**